### PR TITLE
Standardise exception handling

### PR DESCRIPTION
**Pull request by jannaleong**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/PinCommandParser.java [hunk 1]
```diff
@@ -28,11 +25,7 @@ public PinCommand parse(String args) throws ParseException {
 
         ArrayList<String> unknownPrefixes = ArgumentTokenizer.checkUnknownPrefix(args, PREFIX_NAME);
 
-        if (unknownPrefixes.size() > 0) {
-            String exception = String.format(MESSAGE_INVALID_FIELD_FORMAT, unknownPrefixes);
-            exception += "\n" + String.format(MESSAGE_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE);
-            throw new ParseException(exception);
-        }
+        ParserUtil.verifyNoUnknownPrefix(unknownPrefixes, PinCommand.MESSAGE_USAGE);
```
Comment (yleeyilin) [inline src/main/java/seedu/address/logic/parser/PinCommandParser.java:28]: Thanks for helping me catch this!

File: src/main/java/seedu/address/logic/parser/UnpinCommandParser.java [hunk 2]
```diff
@@ -28,11 +25,7 @@ public UnpinCommand parse(String args) throws ParseException {
 
         ArrayList<String> unknownPrefixes = ArgumentTokenizer.checkUnknownPrefix(args, PREFIX_NAME);
 
-        if (unknownPrefixes.size() > 0) {
-            String exception = String.format(MESSAGE_INVALID_FIELD_FORMAT, unknownPrefixes);
-            exception += "\n" + String.format(MESSAGE_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE);
-            throw new ParseException(exception);
-        }
+        ParserUtil.verifyNoUnknownPrefix(unknownPrefixes, UnpinCommand.MESSAGE_USAGE);
```
Comment (yleeyilin) [inline src/main/java/seedu/address/logic/parser/UnpinCommandParser.java:28]: Lovely.

**Review by yleeyilin — APPROVED** [review]
LGTM!

**Review by jamessinmaojun — APPROVED** [review]
lgtm

**Pull request merged**